src/soilbiogeochem/SoilBiogeochemCompetitionMod.F90

```diff
@@ -721,6 +751,76 @@ subroutine SoilBiogeochemCompetition (bounds, num_bgc_soilc, filter_bgc_soilc,nu
                f_n2o_nit_vr(c,j) = f_nit_vr(c,j) * nitrif_n2o_loss_frac
                f_n2o_denit_vr(c,j) = f_denit_vr(c,j) / (1._r8 + n2_n2o_ratio_denit_vr(c,j))
 
+               if (use_soil_nox) then
+
+                  !need to have released N2 to balance the cycle later
+                  f_n2_denit_vr(c,j) = f_denit_vr(c,j)/ (1._r8 + 1._r8/n2_n2o_ratio_denit_vr(c,j))
+ 
+               
+               !soil NOx emissions nitrification
+                  pot_f_nox_nit_vr(c,j)   = f_n2o_nit_vr(c,j) * nox_n2o_ratio_vr(c,j) 
+
+               ! RAIN PULSES FOR NOX NITRIFICATION
+               !An increase of 0.5% (v/v) in 7 cm of surface soil is equivalent to about 3.5 mm of rainfall, which 
+               !is the rainfall amount often reported to cause a pulse
+               ! Calculate the change in volumetric soil moisture since previous timestep 
+                  h2osoi_diff(c,j)=h2osoi_vol(c,j)-h2osoi_vol_old(c,j)
+                  if (h2osoi_vol(c,j)  < 0.30_r8 .and. pfactor_vr(c,j) .eq. 1._r8) then
+
+                  ! If change in soil moisture is > 0.005 (0.5% v/v) and there is precipitation, then start pulsing
+                     if ( h2osoi_diff(c,j) > 0.005_r8 ) then
+                     ! Initialize new pulse factor (dry period hours)
+                        if ( ldry_vr(c,j) > 72._r8 ) then
+                           pfactor_vr(c,j) = 13._r8 *LOG(ldry_vr(c,j)) - 53.6_r8
```

Comment:
This is another large section of the new code. It would be helpful again here to assign hardwired values to variable names (always with literature references when possible).